components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/store/JDBCRecoveryDataStore.java

```diff
@@ -134,6 +135,12 @@ public UserRecoveryData load(User user, Enum recoveryScenario, Enum recoveryStep
     @Override
     public UserRecoveryData load(String code) throws IdentityRecoveryException {
 
+        return load(code, false);
+    }
+
+    @Override
+    public UserRecoveryData load(String code, boolean skipExpiryValidation) throws IdentityRecoveryException, NotImplementedException {
```

Comment:
How are we handling the skipping expiry validation?

Reply:
Code introspection method always calls setting this to 'true'. The rest of the existed use cases always set this to 'false' keeping the existing behavior, which does the expiry validation. There is little out of place login within the Data layer to check the expiry of the codes before return. This parameter is used there to make the switch.